src/app/_components/Form/FormFileInput.tsx

```diff
@@ -0,0 +1,179 @@
+import NextImage from 'next/image'
+
+import {
+  Field,
+  Input,
+  Description,
+  type InputProps as HeadlessInputProps,
+} from '@headlessui/react'
+import { Image, X } from '@phosphor-icons/react/dist/ssr'
+import { clsx } from 'clsx'
+
+import { FormError, type FormErrorProps } from '@/components/Form/FormError'
+import { FormLabel, type FormLabelProps } from '@/components/Form/FormLabel'
+import { Icon } from '@/components/Icon'
+
+type HardCodedProps = 'type' | 'onChange' | 'accept'
+
+export type FormFileInputProps = {
+  files: FileList | null
+  accept: Array<`.${string}`>
+  maxSize: number
+  description?: string | React.ReactNode
+  onChange: (files: FileList | null) => void
+} & Omit<HeadlessInputProps, HardCodedProps> &
+  FormLabelProps &
+  FormErrorProps
+
+export function FormFileInput({
```

Comment:
Maybe we should include keyboard accessibility? Currently you can only add a file by clicking

Reply:
Good idea, done!